src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Strings.vb

```diff
@@ -142,6 +143,92 @@ Namespace Global.Microsoft.VisualBasic
             Return Global.System.Convert.ToChar(CharCode And &HFFFFI)
         End Function
 
+        '============================================================================
+        ' String manipulation functions.
+        '============================================================================
+        Public Function Filter(ByVal Source() As Object, ByVal Match As String, Optional ByVal Include As Boolean = True, <Microsoft.VisualBasic.CompilerServices.OptionCompareAttribute()> Optional ByVal [Compare] As CompareMethod = CompareMethod.Binary) As String()
+
+            Dim Size As Integer = UBound(Source)
+            Dim StringSource(Size) As String
+
+            Try
+                For i As Integer = 0 To Size
+                    StringSource(i) = CStr(Source(i))
+                Next i
+            Catch ex As StackOverflowException
+                Throw ex
+            Catch ex As OutOfMemoryException
+                Throw ex
+            Catch ex As System.Threading.ThreadAbortException
+                Throw ex
+            Catch
+                Throw New ArgumentException(SR.Format(SR.Argument_InvalidValueType2, NameOf(Source), "String"), NameOf(Source))
+            End Try
+
+            Return Filter(StringSource, Match, Include, [Compare])
+        End Function
+
+        Public Function Filter(ByVal Source() As String, ByVal Match As String, Optional ByVal Include As Boolean = True, <Microsoft.VisualBasic.CompilerServices.OptionCompareAttribute()> Optional ByVal [Compare] As CompareMethod = CompareMethod.Binary) As String()
+            Try
+                Dim TmpResult() As String
+                Dim lNumElements As Integer
+                Dim lSourceIndex As Integer
+                Dim lResultIndex As Integer
+                Dim sStringElement As String
+                Dim iFlags As CompareOptions
+                Dim CompInfo As CompareInfo
+                Dim Loc As CultureInfo
+
+                'Do error checking
+                If Source.Rank <> 1 Then
+                    Throw New ArgumentException(SR.Argument_RankEQOne1, NameOf(Source))
+                End If
+
+                If Match Is Nothing OrElse Match.Length = 0 Then
+                    Return Nothing
+                End If
+
+                lNumElements = Source.Length
+
+                'up the globilization info
```

Comment:
Not your typo...

Reply:
Fixed. But as with other comments: do we want to change/refactor the code or keep it in sync with NetFX?